### PR TITLE
feat: refactor HStreamDB bridge to connector and action

### PR DESCRIPTION
**Pull request by kjellwinblad**
This commit also upgrades the hstreamdb_erl driver library and change the action/bridge to use the new hstreamdb_erl.

Much of the code for the new API is copied from:
https://github.com/emqx/emqx-enterprise/blob/be1a1604dd5c4c06ea6be15b702f35d9bdec48a7/lib-ee/emqx_rule_actions/src/emqx_backend_hstreamdb_actions.erl

Fixes:
https://emqx.atlassian.net/browse/EMQX-11458
Release version: v/e5.?

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [] Added property-based tests for code which performs user input validation
- [] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [] Schema changes are backward compatible

### CODE REVIEW

File: rel/i18n/emqx_bridge_hstreamdb.hocon [hunk 1]
```diff
@@ -47,4 +47,58 @@ NOTE: When you use `raw record` template (which means the data is not a valid JS
...
+"""Writer Pool Size"""
+
+batch_size.desc:
+"""Maximum number of insert data clauses that can be sent in a single request"""
+
+batch_size.label:
+"""Max Batch Append Count"""
+
+batch_interval.desc:
+"""Maximum interval in milliseconds that is allowed between two successive (batch) request"""
```
Comment (zmstone) [inline rel/i18n/emqx_bridge_hstreamdb.hocon:87]: duration() is a string like `1s` or `100ms`. there is no need to specify precision in docs.
```suggestion
"""Maximum interval that is allowed between two successive (batch) request."""
```

File: rel/i18n/emqx_bridge_hstreamdb.hocon [hunk 2]
```diff
@@ -47,4 +47,58 @@ NOTE: When you use `raw record` template (which means the data is not a valid JS
...
+"""Max Batches"""
+
+writer_pool_size.desc:
+"""Writer Pool Size"""
+
+writer_pool_size.label:
+"""Writer Pool Size"""
+
+batch_size.desc:
+"""Maximum number of insert data clauses that can be sent in a single request"""
```
Comment (zmstone) [inline rel/i18n/emqx_bridge_hstreamdb.hocon:81]:
```suggestion
"""Maximum number of insert data clauses that can be sent in a single request."""
```

File: rel/i18n/emqx_bridge_hstreamdb.hocon [hunk 3]
```diff
@@ -47,4 +47,58 @@ NOTE: When you use `raw record` template (which means the data is not a valid JS
...
+"""gRPC Flush Period"""
+
+aggregation_pool_size.desc:
+"""Size of Record Aggregation Pool"""
+
+aggregation_pool_size.label:
+"""Aggregation Pool Size"""
+
+max_batches.desc:
+"""Maximum number of unconfirmed batches in the flush queue"""
```
Comment (zmstone) [inline rel/i18n/emqx_bridge_hstreamdb.hocon:69]:
```suggestion
"""Maximum number of unconfirmed batches in the flush queue."""
```

File: rel/i18n/emqx_bridge_hstreamdb.hocon [hunk 4]
```diff
@@ -47,4 +47,58 @@ NOTE: When you use `raw record` template (which means the data is not a valid JS
...
 """HStream Record"""
 
+action_parameters.desc:
+"""Action specific configuration."""
+
+action_parameters.label:
+"""Action"""
+
+grpc_flush_timeout.desc:
+"""Period for flushing gRPC calls to the HStreamDB server"""
```
Comment (zmstone) [inline rel/i18n/emqx_bridge_hstreamdb.hocon:57]:
```suggestion
"""Time interval for flushing gRPC calls to the HStreamDB server."""
```

File: rel/i18n/emqx_bridge_hstreamdb.hocon [hunk 5]
```diff
@@ -47,4 +47,58 @@ NOTE: When you use `raw record` template (which means the data is not a valid JS
...
+
+grpc_flush_timeout.label:
+"""gRPC Flush Period"""
```
Comment (zmstone) [inline rel/i18n/emqx_bridge_hstreamdb.hocon:60]:
```suggestion
"""gRPC Flush Interval"""
```

File: rel/i18n/emqx_bridge_hstreamdb.hocon [hunk 6]
```diff
@@ -47,4 +47,58 @@ NOTE: When you use `raw record` template (which means the data is not a valid JS
...
+
+aggregation_pool_size.desc:
+"""Size of Record Aggregation Pool"""
```
Comment (zmstone) [inline rel/i18n/emqx_bridge_hstreamdb.hocon:63]:
```suggestion
"""Size of record aggregation pool."""
```

File: rel/i18n/emqx_bridge_hstreamdb.hocon [hunk 7]
```diff
@@ -47,4 +47,58 @@ NOTE: When you use `raw record` template (which means the data is not a valid JS
...
+"""Aggregation Pool Size"""
+
+max_batches.desc:
+"""Maximum number of unconfirmed batches in the flush queue"""
+
+max_batches.label:
+"""Max Batches"""
+
+writer_pool_size.desc:
+"""Writer Pool Size"""
```
Comment (zmstone) [inline rel/i18n/emqx_bridge_hstreamdb.hocon:75]:
```suggestion
"""The size of the writer pool."""
```

Comment (zmstone) [reply rel/i18n/emqx_bridge_hstreamdb.hocon:75]: might need more information about what it means for performance etc.

File: rel/i18n/emqx_bridge_hstreamdb.hocon [hunk 8]
```diff
@@ -47,4 +47,58 @@ NOTE: When you use `raw record` template (which means the data is not a valid JS
...
+"""Max Batch Append Count"""
+
+batch_interval.desc:
+"""Maximum interval in milliseconds that is allowed between two successive (batch) request"""
+
+batch_interval.label:
+"""Max Batch Interval"""
+
+hstreamdb_action.desc:
+"""Configuration for HStreamDB Action"""
```
Comment (zmstone) [inline rel/i18n/emqx_bridge_hstreamdb.hocon:93]:
```suggestion
"""Configuration for HStreamDB action."""
```

File: rel/i18n/emqx_bridge_hstreamdb.hocon (hunk 6 above)
Comment (zmstone) [reply rel/i18n/emqx_bridge_hstreamdb.hocon:63]: might need more information. e.g. what does it mean to set a larger pool or smaller pool.

**Review by zmstone — APPROVED** [review]

File: apps/emqx_bridge_hstreamdb/src/emqx_bridge_hstreamdb.erl [hunk 9]
```diff
@@ -1,15 +1,17 @@
-%%--------------------------------------------------------------------
+%--------------------------------------------------------------------
```
Comment (HJianBo) [inline apps/emqx_bridge_hstreamdb/src/emqx_bridge_hstreamdb.erl:1]: nit

File: apps/emqx_bridge_hstreamdb/src/emqx_bridge_hstreamdb_connector.erl [hunk 10]
```diff
@@ -140,142 +210,152 @@ desc(config) ->
...
+            ?SLOG(error, #{
+                msg => "hstreamdb_connector_client_not_alive",
+                connector => InstId,
+                error => Error
+            }),
+            {error, {connect_failed, Error}}
+    end.
+
+client_options(Config = #{url := ServerURL, ssl := SSL}) ->
+    GRPCTimeout = maps:get(<<"grpc_timeout">>, Config, ?DEFAULT_GRPC_TIMEOUT),
```
Comment (HJianBo) [inline apps/emqx_bridge_hstreamdb/src/emqx_bridge_hstreamdb_connector.erl:247]:
```
    GRPCTimeout = maps:get(grpc_timeout, Config, ?DEFAULT_GRPC_TIMEOUT),
```

**Pull request merged**